src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,207 @@
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  int effectiveCoordinate = _axis;
+  PRECICE_ASSERT(effectiveCoordinate == 0 ||
+                     effectiveCoordinate == 1 ||
+                     effectiveCoordinate == 2,
+                 "Unknown multiscale axis type.")
+
+  // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+  Eigen::VectorXd axisMidpoints(inSize);
+  _axisMidpoints        = axisMidpoints;
+  auto ordered_vertices = input()->vertices();
+  std::sort(ordered_vertices.begin(), ordered_vertices.end(),
+            [effectiveCoordinate](const mesh::Vertex &a, const mesh::Vertex &b) {
+              return a.getCoords()[effectiveCoordinate] < b.getCoords()[effectiveCoordinate];
+            });
+  for (size_t i = 0; i < (inSize - 1); i++) {
+    auto axisPositionCurrent = ordered_vertices[i].getCoords()[effectiveCoordinate];
+    auto axisPositionNext    = ordered_vertices[i + 1].getCoords()[effectiveCoordinate];
+    _axisMidpoints(i)        = (axisPositionCurrent + axisPositionNext) / 2;
+  }
+  _axisMidpoints(inSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == SPREAD) {
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to the nearest neighbors of the 1D vertices in projection space.
+      */
+      for (size_t i = 0; i < outSize; i++) {
+        auto vertexCoord = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index       = 0;
+        while (vertexCoord > _axisMidpoints(index)) {
+          index++;
+          PRECICE_ASSERT(index < static_cast<int>(inSize));
+        }
+        _vertexIndicesSpread.push_back(index);
+      }
+    } else {
+      PRECICE_ASSERT(_type == COLLECT)
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to (and averaged at) the nearest 1D vertex in projection space.
+      */
+      size_t const inSize  = input()->vertices().size();
+      size_t const outSize = output()->vertices().size();
+
+      // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+      Eigen::VectorXd axisMidpoints(outSize);
+      _axisMidpoints = axisMidpoints;
+      for (size_t i = 0; i < (outSize - 1); i++) {
+        auto axisPositionCurrent = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        auto axisPositionNext    = output()->vertices()[i + 1].getCoords()[effectiveCoordinate];
+        _axisMidpoints(i)        = (axisPositionCurrent + axisPositionNext) / 2;
+      }
+      _axisMidpoints(outSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
```

Comment:
Defining the midpoints seems to be duplicate in both versions of the mapping. Maybe move this outside the if-branch? The assignment to segments also seems to be duplicate.

I can also try merging the different cases myself later, if this is not trivial.

Reply:
In my current version, the `axisMidpoint` assignment happens before 
```
if (getConstraint() == CONSISTENT) {
    PRECICE_DEBUG("Compute consistent mapping");
    if (_type == SPREAD) {` //etc.
```

Reply:
> But I forgot to delete the duplice in the `COLLECT` branch. Did that now.

Should there be any non-pushed changes to this file? I am a bit confused.

Reply:
Looking at it again, I understand that duplication makes (at least in a first view) sense here: in the first case, we are defining the midpoints of the input mesh, in the second case, of the output mesh.

I moved the definitions of `inSize` and `outSize` outside of the `if`, but moved the first case of computing midpoints (of the input mesh) inside the `SPREAD` case. Also added a step to order the vertices of the output mesh in the `COLLECT` case.